tmva/tmva/src/MethodDL.cxx

```diff
@@ -342,6 +345,11 @@ void MethodDL::ProcessOptions()
          settings.regularization = DNN::ERegularization::kL2;
       }
 
+      TString optimizer = fetchValueTmp(block, "Optimizer", TString("SGD"));
+      if (optimizer == "SGD") {
+         settings.optimizer = DNN::EOptimizer::kSGD;
```

Comment:
I guess here the "else part" can never be reached because if the optimizer option is not specified in the training strategy string, the default value "SGD" will be taken.

Reply:
But what if I misspell the optimizer as `"SDG"`? *Please* initialize the member, ideally using a member initializer!

Reply:
Yeah! That would cause a problem in case of misspelling. I'll change that.